StandaloneMmPkg/Library/StandaloneMmMemLib/X86StandaloneMmMemLibInternal.c

```diff
@@ -135,3 +138,199 @@ MmMemLibInternalFreeMmramRanges (
     FreePool (mMmMemLibInternalMmramRanges);
   }
 }
+
+/**
+  Merge continuous memory map entries whose type is
+  EfiLoaderCode/Data, EfiBootServicesCode/Data, EfiConventionalMemory,
+  EfiUnusableMemory, EfiACPIReclaimMemory, because the memory described by
+  these entries will be set as NOT present in SMM page table.
+
+  @param[in, out]  MemoryMap              A pointer to the buffer in which firmware places
+                                          the current memory map.
+  @param[in, out]  MemoryMapSize          A pointer to the size, in bytes, of the
+                                          MemoryMap buffer. On input, this is the size of
+                                          the current memory map.  On output,
+                                          it is the size of new memory map after merge.
+  @param[in]       DescriptorSize         Size, in bytes, of an individual NON_MM_MEMORY_MAP.
+**/
+STATIC
+VOID
+MergeOverlappedOrContinuousRanges (

```

Comment:
can we comments why need this merge? i remember we need this action due to 2 cases we discussed before?

Reply:
updated comments here